hpx/runtime/parcelset/detail/call_for_each.hpp

```diff
@@ -45,12 +45,13 @@ namespace hpx { namespace parcelset
                 return *this;
             }
 
-            void operator()(boost::system::error_code const& e) const
+            void operator()(boost::system::error_code const& e)
             {
                 HPX_ASSERT(parcels_.size() == handlers_.size());
                 for(std::size_t i = 0; i < parcels_.size(); ++i)
                 {
                     handlers_[i](e, parcels_[i]);
+                    handlers_[i].reset();
                 }
```

Comment:
Is this change related to this PR?

Reply:
Not directly... Not sure why I included it. Potentially, it leads to resources being freed in a timelier fashion. I could add it in a different PR if you wish.